docs/setuptools.txt

```diff
@@ -2087,16 +2087,8 @@ New in 41.5.0: Deprecated the test command.
 ``upload`` - Upload source and/or egg distributions to PyPI
 ===========================================================
 
-.. warning::
-    **upload** is deprecated in favor of using `twine
-    <https://pypi.org/p/twine>`_
-
-The ``upload`` command is implemented and `documented
-<https://docs.python.org/3.1/distutils/uploading.html>`_
-in distutils.
-
-New in 20.1: Added keyring support.
-New in 40.0: Deprecated the upload command.
+The ``upload`` command was deprecated in version 40.0 and removed in version
+42.0. Use `twine <https://pypi.org/p/twine>`_ instead.
```

Comment:
I think it'd be helpful to show folks exactly what command they should use instead here, something like "Instead of `python setup.py upload`, use `twine upload dist/*`"

Reply:
Agreed, except those commands aren't equivalent. `setup.py upload` would have no effect unless preceded with a `_dist` command... whereas `twine` requires rethinking your workflow (first run your `_dist` command or a `pep517` builder, then run twine on the output of that command). Do we really want to try to describe all those details, or leave it vague and let the user's rely on the twine docs? I lean toward the latter.

Reply:
Given how infrequently we update this documentation, I think a link is probably better than an explanation (especially since we actually already have an explanation on how to use `twine` earlier in this same document). I updated the documentation to point at the Python Packaging User Guide.

Sounds good?

Reply:
I think that's a good compromise.